rsocket-core/src/main/java/io/rsocket/loadbalance/StatsRequestInterceptor.java

```diff
@@ -0,0 +1,87 @@
+package io.rsocket.loadbalance;
+
+import io.netty.buffer.ByteBuf;
+import io.rsocket.frame.FrameType;
+import io.rsocket.plugins.RequestInterceptor;
+import reactor.util.annotation.Nullable;
+
+public class StatsRequestInterceptor implements RequestInterceptor {
```

Comment:
Maybe `WeightedStatsRequestInterceptor` and likewise `Stats` --> `WeightedStats` since those are associated with the `WeightedLoadbalanceStrategy`.

Reply:
fixed